### DISCUSSION TITLE
Optimizing the obtaining of the data of the video to be downloaded

### DISCUSSION THREAD

**Discussion by BobbyESP** [JunkFood02/Seal#105] [Ideas]
I've been thinking on this and i have an idea to improve the fetching of the video information to the app be more efficient at the downloading time.
We can implement that when the TextBox is modified with a valid link, inmediately it will start adquiring the video information in the background so when clicked in the download button, the download provess will start without adquiring another time the video data.

**Comment by JunkFood02** [comment]
Sending a network request when the user isn't taking an operation seems a bit annoying. Also, we're not able to know if a link is "valid" before fetching its info.

**Reply by BobbyESP** [reply]
Well, this can be an option that we can put in settings and for knowing if it's a valid link, we can use Regex

**Reply by JunkFood02** [reply]
> Well, this can be an option that we can put in settings and for knowing if it's a valid link, we can use Regex

We already use regex for matching links in the clipboard, but a valid link should contain a downloadable video.

**Reply by BobbyESP** [reply]
Yeah... I'm going to think another method so. Thank you for your time!
